homeassistant/components/gios/config_flow.py

```diff
@@ -0,0 +1,73 @@
+"""Adds config flow for GIOS."""
+import asyncio
+
+from aiohttp.client_exceptions import ClientConnectorError
+from async_timeout import timeout
+from gios import ApiError, Gios, NoStationError
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_STATION_ID, DEFAULT_NAME, DOMAIN  # pylint:disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_STATION_ID): int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+    }
+)
+
+
+class GiosFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for GIOS."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                already_configured = self._async_current_ids()
```

Comment:
You could use self._abort_if_unique_id_configured() here

Reply:
Done